pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -214,40 +250,233 @@ func (c *MCRouteController) Run(stopCh <-chan struct{}) {
 	}
 
 	for i := 0; i < defaultWorkers; i++ {
-		go wait.Until(c.worker, time.Second, stopCh)
+		go wait.Until(c.gatewayWorker, time.Second, stopCh)
+		go wait.Until(c.ciImpWorker, time.Second, stopCh)
 	}
 	<-stopCh
 }
 
-// worker is a long-running function that will continually call the processNextWorkItem
-// function in order to read and process a message on the workqueue.
-func (c *MCRouteController) worker() {
-	for c.processNextWorkItem() {
+// gatewayWorker is a long-running function that will continually call the processNextGatewayItem
+// function in order to read and process a message on the gatewayQueue.
+func (c *MCRouteController) gatewayWorker() {
+	for c.processNextGatewayItem() {
 	}
 }
 
-func (c *MCRouteController) processNextWorkItem() bool {
-	obj, quit := c.queue.Get()
+// ciImpWorker is a long-running function that will continually call the processNextClusterInfoImportItem
+// function in order to read and process a message on the ciImpQueue.
+func (c *MCRouteController) ciImpWorker() {
+	for c.processNextClusterInfoImportItem() {
+	}
+}
+
+func (c *MCRouteController) processNextClusterInfoImportItem() bool {
+	obj, quit := c.ciImpQueue.Get()
 	if quit {
 		return false
 	}
-	defer c.queue.Done(obj)
+	defer c.ciImpQueue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
-		c.queue.Forget(obj)
+	ciImp, ok := obj.(string)
+	if !ok {
+		c.ciImpQueue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+	if c.wireGuardConfig != nil {
+		if err := c.updateWireGuardPeers(ciImp); err != nil {
```

Comment:
Does `updateWireGuardPeers` support reentrance (e.g. the func fails partially, or `syncMCFlows` fails, and triggers a retry)?

Please revisit all these cases and probably add a comment to the funcs need to support reentrance.

Reply:
Thanks for the reminder, revisited the cases and add some checks for reentrance.